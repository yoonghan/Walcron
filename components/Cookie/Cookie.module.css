.container {
  width: 100%;
  bottom: 0;
  position: fixed;
  background-color: rgba(50, 50, 100, 0.9);
  color: var(--text, #fff);
  display: flex;
  padding: 20px;
  z-index: 999;
}

.container p {
  padding: 0;
}

.message {
  width: 75%;
  font-size: 0.9rem;
}
.button-container {
  width: 80%;
  text-align: center;
  align-self: center;
}
.button-container > button {
  border-radius: 5px;
  cursor: pointer;
  padding: 1rem;
  background-color: #fff;
  color: rgba(50, 50, 100, 0.9);
  font-weight: bold;
}

@media only screen and (max-width: 480px) {
  .container {
    padding: 2px 10px;
    flex-direction: column;
  }

  .message {
    width: 100%;
  }

  .button-container {
    align-self: center;
  }

  .button-container > button {
    width: 100%;
    align-self: flex-end;
    color: #00b4ef;
    border: none;
  }
}
