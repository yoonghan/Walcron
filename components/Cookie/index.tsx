`use strict`

/**
* Made for EUROPEAN regulation, but this site do not use cookies.
* For references, pages/* need to provide the context in getInitialProps and
* pass into the props.
**/

import * as React from "react";
import {COOKIEBOX} from "../../shared/style";

interface CookieProps {
  isClosed:boolean;
}


export const wasCookieRead = () => {
  const cookie = document.cookie;

  if(typeof cookie === "undefined") {
    return false;
  }

  const keyValues = cookie.split(";");
  const filterTermsValue = keyValues.filter(keyValue => {
    return keyValue.startsWith("termsRead");
  });

  return filterTermsValue && (filterTermsValue[0].split("=")[1] == "true");
}

const Cookie:React.FC<CookieProps> = (props) => {

  const [isClosed, setClosed] = React.useState(props.isClosed || false);

  const reactToCookieButton = () => {
    setClosed(true);
    saveCookieWithTerms();
  }

  const saveCookieWithTerms = () => {
    document.cookie = "termsRead=true";
  }

  if(isClosed) {
    return <React.Fragment/>;
  }

  return (
    <div className={"container"}>
      <div className={"message"}>
        <h4>This site uses cookies.</h4>
        <p>Your generated cookie, shall and will not be used for
        any public sharing or advertisement related purposes. The cookie is only
        saved in your local and private browser to monitor visits and usage traffics.</p>
        <p>By proceeding on this website, you have indirectly accepted and agreed to our cookie usage.</p>
      </div>
      <div className={"button-container"}>
        <button onClick={reactToCookieButton} >Close</button>
      </div>
      <style jsx>{`
        .container{
          width: 100%;
          bottom: 0;
          position: fixed;
          background-color: ${COOKIEBOX.BACKGROUND};
          color: ${COOKIEBOX.FOREGROUND};
          display: flex;
          padding: 20px;
        }
        .message {
          width: 75%;
          font-size: 0.9rem;
        }
        .button-container {
          width: 25%;
          text-align: center;
          align-self: center;
        }
        .button-container > button {
          border-radius: 5px;
          cursor: pointer;
          padding: 1rem;
          background-color: ${COOKIEBOX.FOREGROUND};
          color: ${COOKIEBOX.BACKGROUND};
          font-weight: bold;
        }
        @media only screen and (max-width: 480px) {
          .container{
            height: 150px;
            padding: 2px 10px;
          }
        }
      `}</style>
    </div>
  );
}

export default Cookie;
