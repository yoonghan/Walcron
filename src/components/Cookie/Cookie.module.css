.container {
  width: 100%;
  bottom: 0;
  position: fixed;
  background-color: rgba(50, 50, 100, 0.9);
  color: var(--text, #fff);
  display: flex;
  padding: 20px;
  z-index: 999;
}

.title {
  font-size: 1rem;
  font-weight: bold;
}

.container p {
  padding: 0;
}

.message {
  width: 75%;
  font-size: 0.9rem;
  flex: 2;
}
.button-container {
  width: 80%;
  text-align: center;
  align-self: center;
  flex: 1;
}
.button-container > button {
  border-radius: 5px;
  cursor: pointer;
  padding: 1rem;
  background-color: #fff;
  font-weight: bold;
}

@media only screen and (max-width: 480px) {
  .container {
    padding: 2px 10px;
    flex-direction: column;
    width: auto;
  }

  .message {
    width: 100%;
    flex: 1;
  }

  .button-container {
    align-self: center;
  }

  .button-container > button {
    width: 100%;
    align-self: flex-end;
    border: none;
  }
}
