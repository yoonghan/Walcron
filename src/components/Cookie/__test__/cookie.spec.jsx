import * as React from 'react';
import { shallow } from 'enzyme';
import Cookie from "../index";

describe('ButtonsBar', () => {
  it('renders closed normally', () => {
    const wrapper = shallow(
      <Cookie isClosed={true}/>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  }),
  it('renders open normally', () => {
    const wrapper = shallow(
      <Cookie isClosed={false}/>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
