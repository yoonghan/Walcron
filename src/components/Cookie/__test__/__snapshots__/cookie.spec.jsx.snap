// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsBar renders closed normally 1`] = `"<Fragment />"`;

exports[`ButtonsBar renders open normally 1`] = `
"<div className=\\"jsx-3717058259 container\\">
  <div className=\\"jsx-3717058259 message\\">
    <h4 className=\\"jsx-3717058259\\">
      This site uses cookies.
    </h4>
    <p className=\\"jsx-3717058259\\">
      This site uses cookie to monitor visits and usage traffics. We use google analytics, please 
      <a href=\\"https://policies.google.com/\\" target=\\"_blank\\" className=\\"jsx-3717058259\\">
         refer here
      </a>
      .
    </p>
    <p className=\\"jsx-3717058259\\">
      By proceeding on this website, you are accepting and agreed to the cookie usage.
    </p>
  </div>
  <div className=\\"jsx-3717058259 button-container\\">
    <button onClick={[Function: reactToCookieButton]} className=\\"jsx-3717058259\\">
      Close
    </button>
  </div>
  <JSXStyle id=\\"585578432\\" dynamic={{...}}>
    a.__jsx-style-dynamic-selector{-webkit-text-decoration:underline;text-decoration:underline;}.container.__jsx-style-dynamic-selector{width:100%;bottom:0;position:fixed;background-color:rgba(50,50,100,0.9);color:#FFF;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding:20px;}.message.__jsx-style-dynamic-selector{width:75%;font-size:0.9rem;}.button-container.__jsx-style-dynamic-selector{width:25%;text-align:center;-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}.button-container.__jsx-style-dynamic-selector&gt;button.__jsx-style-dynamic-selector{border-radius:5px;cursor:pointer;padding:1rem;background-color:#FFF;color:rgba(50,50,100,0.9);font-weight:bold;}@media only screen and (max-width:480px){.container.__jsx-style-dynamic-selector{padding:2px 10px;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.button-container.__jsx-style-dynamic-selector{-webkit-align-self:flex-end;-ms-flex-item-align:end;align-self:flex-end;}.button-container.__jsx-style-dynamic-selector&gt;button.__jsx-style-dynamic-selector{-webkit-align-self:flex-end;-ms-flex-item-align:end;align-self:flex-end;background:none;color:unset;border:none;}}
  </JSXStyle>
</div>"
`;
